app/models/account/gateway.rb

```diff
@@ -3,7 +3,7 @@ module Account::Gateway
 
   included do
     has_many :payment_transactions
-    has_one :payment_gateway_setting, dependent: :destroy, inverse_of: :account
+    has_one :payment_gateway_setting, dependent: :destroy, inverse_of: :account, class_name: 'PaymentGatewaySetting'
```

Comment:
Why adding the `class_name`. `'PaymentGatewaySetting'` is what Rails will infer by default, no?

Reply:
Because otherwise it is `nil` and doesn't go through `DeletePaymentSettingHierarchyWorker` but `DeleteObjectHierarchyWorker` instead.
That is in the tests so you can see it if you remove this change and run the new tests :smile: